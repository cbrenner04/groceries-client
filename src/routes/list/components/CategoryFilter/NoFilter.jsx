import React from 'react';
import { Button } from 'react-bootstrap';

const NoFilter = () => (
  <Button variant="light" disabled style={{ cursor: 'not-allowed' }}>
    Filter by category
  </Button>
);

export default NoFilter;
