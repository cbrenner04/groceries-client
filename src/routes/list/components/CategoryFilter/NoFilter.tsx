import React from 'react';
import { Button } from 'react-bootstrap';

const NoFilter: React.FC = () => (
  <Button variant="light" id="no-filter" disabled>
    Filter by category
  </Button>
);

export default NoFilter;
