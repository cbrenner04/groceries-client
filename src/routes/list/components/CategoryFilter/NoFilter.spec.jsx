import React from 'react';
import { render } from '@testing-library/react';

import NoFilter from './NoFilter';

describe('NoFilter', () => {
  it('renders a disabled button', async () => {
    const { container, findByRole } = render(<NoFilter />);

    expect(container).toMatchSnapshot();
    expect(await findByRole('button')).toBeDisabled();
  });
});
