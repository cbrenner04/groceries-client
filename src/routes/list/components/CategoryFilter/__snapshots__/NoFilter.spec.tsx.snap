// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoFilter renders a disabled button 1`] = `
<div>
  <button
    class="btn btn-light"
    disabled=""
    id="no-filter"
    type="button"
  >
    Filter by category
  </button>
</div>
`;
